tests/installation/select_patterns_and_packages.pm

```diff
@@ -155,113 +141,128 @@ sub package_action {
-                    }
-                }
-            }
-            $needs_to_be_selected = 1 if ($patterns{all});
+sub select_specific_patterns_by_iteration {
+    my %patterns;    # set of patterns to be processed
+                     # fill with variable values
+    @patterns{split(/,/, get_var('PATTERNS'))} = ();
 
-            my $selected = check_screen([qw(current-pattern-selected on-category)], 0);
+    my $counter = 80;
+    # delete special 'all' and 'default' keys from the check
+    delete $patterns{default};
+    delete $patterns{all};
+    while (1) {
+        die "looping for too long" unless ($counter--);
+        my $needs_to_be_selected;
+        my $ret = check_screen('on-pattern', 1);
+        # this variable will only be updated when the pattern list entry is under the cursor
+        # and the pattern is in the PATTERNS variable and there is a needle for that entry
+        my $current_pattern = 'UNKNOWN_PATTERN';
 
-            if ($selected && $selected->{needle}->has_tag('on-category')) {
-                movedownelseend;
-                check12qtbug;
-                next;
-            }
-            if ($needs_to_be_selected && !$selected) {
-                wait_screen_change {
-                    send_key ' ';
-                    record_info("select", "");
-                };
-                assert_screen 'current-pattern-selected', 5;
+        if ($ret) {    # unneedled pattern
+            for my $p (keys %patterns) {
+                my $sel     = 1;
+                my $pattern = $p;    # store pattern untouched
+                if ($p =~ /^-/) {
+                    # this pattern shall be deselected as indicated by '-' prefix
+                    $sel = 0;
+                    $p =~ s/^-//;
+                }
+                if (match_has_tag("pattern-$p")) {
+                    $needs_to_be_selected = $sel;
+                    $current_pattern      = $p;
+                    delete $patterns{$pattern};
+                    record_info($current_pattern, $needs_to_be_selected);
+                }
             }
-            # stick to the default patterns. Check if at least 1 dep. issue was displayed
-            $dep_issue = $self->workaround_dependency_issues || $dep_issue;
+            last if ((get_var('PATTERNS', '') =~ /default/) && !(scalar keys %patterns));
+        }
 
-            if (get_var('PATTERNS', '') =~ /default/ && !(get_var('PATTERNS', '') =~ /$current_pattern/)) {
-                $needs_to_be_selected = $selected;
-                record_info("keep default", "");
-            }
-            if (!$needs_to_be_selected && $selected) {
-                send_key ' ';
-                record_info("unselect", "");
-                assert_screen [qw(current-pattern-unselected current-pattern-autoselected)], 8;
-            }
-            movedownelseend;
-            check12qtbug;
+        $needs_to_be_selected = 1 if get_var('PATTERNS', '') =~ /all/;
```

Comment:
As we use different way to select all patterns, I guess we can remove this line as is handled in `select_all_patterns_by_menu`. Or am I missing something?

Reply:
I wasn't sure if some text scenario could need it... due to line #239.

Reply:
Ah, you're totally right!